clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/RateSensor.java

```diff
@@ -32,15 +38,27 @@ public double getRate() {
    * Record the rate value.
    */
   public void record() {
-    sensor.record();
+    rateAccumulator++;
```

Comment:
Is it better to use `AtomicLong::getAndIncrement` and `AtomicLong::getAndAdd` here rather than `++`?  
If this can be called simultaneously by multiple threads then `++` can make the per sec value very misleading.

Reply:
`AtomicLong` is expected to work well only in low contention scenarios, but not in high contention... and a naked ++ has correctness issues as you pointed out. Let's hold off on this change for now.

Reply:
given `record` already uses synchronized,  if we use synchronized in the increment/add part will that be better than current code?

Reply:
Exactly... it won't be better, just shift the problem elsewhere.

Reply:
In high contention scenarios, how atomic variables might perform relative to the current `synchronized` block? Will it be worth it to test the below to see how it performs against the current `synchronized` block?
1. use `AtomicLong` for aggregating per sec
2. use `CAS` for calling the `sensor.record()` with the above aggregated metrics which should not block other threads.

Reply:
just a to test the theory, I benchmarked a synchronized vs atomic int counter code with some 100 threads counting to 1000. `synchronized` performed much better than atomic, so the theory is correct that highly contended use case atomics are not good. 
 AtomicLong    253323.714 ± 1157589.681   ns/op
 synchronized  180800.700 ± 229209.223   ns/op

Reply:
Given today tehuti calls Time.millisecords and synchronized on a much bigger block of code on every `record` call, the proposed solution of using a synchronized on add/increment and time call every second would still be better, no?